File: backend/app/schemas/brand.py
"""品牌请求/响应模型"""
from pydantic import BaseModel


class BrandCreate(BaseModel):
    """创建品牌请求"""
    name: str
    country: str
    logo_url: str | None = None
    description: str | None = None


class BrandUpdate(BaseModel):
    """更新品牌请求"""
    name: str | None = None
    country: str | None = None
    logo_url: str | None = None
    description: str | None = None


class BrandResponse(BaseModel):
    """品牌响应"""
    id: int
    name: str
    country: str
    logo_url: str | None
    description: str | None
    juice_count: int = 0
    created_at: str

File: frontend/src/api/brand.ts
import client from './client';

export interface BrandItem {
  id: number;
  name: string;
  country: string;
  logo_url: string | null;
  description: string | null;
  juice_count: number;
  created_at: string;
}

export interface BrandListData {
  items: BrandItem[];
  total: number;
  page: number;
  size: number;
}

export const getBrands = (params: { page?: number; size?: number; country?: string }) =>
  client.get<{ code: number; data: BrandListData }>('/brands', { params });

export const getBrand = (id: number) =>
  client.get<{ code: number; data: BrandItem }>(`/brands/${id}`);
